server.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/google/go-github/github"
+	pb "github.com/itachigit/github-search-service/githubsearchpb"
+
+	"google.golang.org/grpc"
+)
+
+type server struct {
+	pb.UnimplementedGithubSearchServiceServer
+}
+
+func (s *server) Search(ctx context.Context, req *pb.SearchRequest) (*pb.SearchResponse, error) {
+	query := req.SearchTerm
+	if req.User != "" {
+		query += fmt.Sprintf(" user:%s", req.User)
+	}
+
+	apiURL := os.Getenv("GITHUB_API_URL")
+	if apiURL == "" {
+		apiURL = "https://api.github.com"
+	}
+
+	url := fmt.Sprintf("%s/search/code?q=%s", apiURL, url.QueryEscape(query))
+
+	reqHttp, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	token := os.Getenv("GITHUB_TOKEN")
+	if token != "" {
+		reqHttp.Header.Set("Authorization", "Bearer "+token)
+	}
+
+	resp, err := http.DefaultClient.Do(reqHttp)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if !(resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusMultipleChoices) {
```

Comment:
Using raw int constants here will be more readable.
```suggestion
	if !(resp.StatusCode >= 200 && resp.StatusCode < 300) {
```